### PR TITLE
mtl/ofi: add some more mca parameters

### PR DESCRIPTION
**Pull request by hppritcha**
allow for toggling of both control/data progress models.
allow for using FI_AV_TABLE or FI_AV_MAP for av type.

@yburette 

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

### CODE REVIEW

**Comment by yburette** [comment]
@hppritcha Sounds good to me. Are these options required by the gni provider? Are there any other modifications needed to make it work?

File: ompi/mca/mtl/ofi/mtl_ofi_component.c [hunk 1]
```diff
@@ -82,6 +85,33 @@ ompi_mtl_ofi_component_register(void)
...
+    av_type = "map";
+    mca_base_component_var_register(&mca_mtl_ofi_component.super.mtl_version,
+                                    "av_type",
+                                    "Specify AV type to use (default: map). Set to table for FI_AV_TABLE AV type.",
+                                    MCA_BASE_VAR_TYPE_STRING, NULL, 0, 0,
+                                    OPAL_INFO_LVL_3,
+                                    MCA_BASE_VAR_SCOPE_READONLY,
+                                    &av_type);
+
     return OMPI_SUCCESS;
```
Comment (hjelmn) [inline ompi/mca/mtl/ofi/mtl_ofi_component.c]: @hppritcha This is a good place to use the mca_base_var_enum class. You can let the variable system handle the conversion of string->int.

**Comment by hppritcha** [comment]
@yburette these are not required by the GNI provider but testers requested them to allow for easier triaging of potential bugs.

**Comment by yburette** [comment]
@hppritcha Fair enough!

**Comment by hppritcha** [comment]
@hjelmn take a look now

**Pull request merged**